It fixes Issue 13.

Machiry opened the pull request:
Github Url: https://github.com/linaro-swg/linux/issues/13

Zeroing out Shared memory allocated by tee linux kernel driver.

Machiry committed:
It fixes Issue 13.

Github Url: https://github.com/linaro-swg/linux/issues/13

Zeroing out Shared memory allocated by tee linux kernel driver.

jockebech commented:
We need a better commit message. Just changing the order would make more sense, like:

```
optee: Zeroing out shared memory

To prevent information leakage, we are clearing the shared memory
buffers that has been allocated by the TEE Linux kernel driver.

Fixes this GitHub issue:
https://github.com/linaro-swg/linux/issues/13
```

Other than the commit message:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

jenswikl commented on drivers/tee/tee_shm_pool.c:
```diff
@@ -28,6 +28,7 @@ static int pool_op_gen_alloc(struct tee_shm_pool_mgr *poolm,
 	va = gen_pool_alloc(genpool, s);
 	if (!va)
 		return -ENOMEM;
+	memset(va, 0, s);
```
Surely this must result in a warning since `va` is an `unsigned long` and not a pointer.

Machiry committed:
optee: Zeroing out shared memory

To prevent information leakage, we are clearing the shared memory
buffers that has been allocated by the TEE Linux kernel driver.

Fixes this GitHub issue:
https://github.com/linaro-swg/linux/issues/13

jenswi-linaro commented:
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

jockebech commented:
@jenswi-linaro did I read somewhere that you'd like to just add this into the coming patches on your own or should I just go ahead and merge it?

jenswi-linaro commented:
Please merge, I'll pick it up later.

ghost commented:
Thanks for the fix @Machiry, closing the PR since it has been merged manually.
